File: LeetCode/0190. Reverse Bits/solution.cpp
/**
 * 600 / 600  test cases passed.
 * Runtime: 0 ms
 * Memory Usage: 5.9 MB 
 */
class Solution {
public:
    uint32_t reverseBits(uint32_t n) {
        uint32_t ans = 0;
        for (int i = 32; i --> 0;) {
            ans = ans << 1 | n & 1;
            n >>= 1;    
        }
        return ans;
    }
};

File: LeetCode/0190. Reverse Bits/solution.py
"""
600 / 600 test cases passed.
Runtime: 36 ms
Memory Usage: 14.7 MB
"""
class Solution:
    def reverseBits(self, n: int) -> int:
        ans = 0
        for i in range(32):
            ans = ans << 1 | n & 1
            n >>= 1
        return ans
